File: app/src/main/java/wangdaye/com/geometricweather/background/polling/permanent/update/ForegroundTodayForecastUpdateService.java
package wangdaye.com.geometricweather.background.polling.permanent.update;

import android.content.Context;

import androidx.core.app.NotificationCompat;

import wangdaye.com.geometricweather.GeometricWeather;
import wangdaye.com.geometricweather.R;
import wangdaye.com.geometricweather.background.polling.basic.ForegroundUpdateService;
import wangdaye.com.geometricweather.basic.model.location.Location;
import wangdaye.com.geometricweather.remoteviews.presenter.notification.ForecastNotificationIMP;

/**
 * Foreground Today forecast update service.
 * */

public class ForegroundTodayForecastUpdateService extends ForegroundUpdateService {

    @Override
    public void updateView(Context context, Location location) {
        if (ForecastNotificationIMP.isEnable(this, true)) {
            ForecastNotificationIMP.buildForecastAndSendIt(context, location, true);
        }
    }

    @Override
    public void handlePollingResult(boolean failed) {
        // do nothing.
    }

    @Override
    public NotificationCompat.Builder getForegroundNotification(int index, int total) {
        return super.getForegroundNotification(index, total)
                .setContentTitle(getString(R.string.geometric_weather) + " " + getString(R.string.forecast));
    }

    @Override
    public int getForegroundNotificationId() {
        return GeometricWeather.NOTIFICATION_ID_UPDATING_TODAY_FORECAST;
    }
}

File: app/src/main/java/org/breezyweather/background/polling/services/permanent/update/ForegroundTodayForecastUpdateService.kt
package org.breezyweather.background.polling.services.permanent.update

import android.content.Context
import androidx.core.app.NotificationCompat
import dagger.hilt.android.AndroidEntryPoint
import org.breezyweather.R
import org.breezyweather.background.polling.services.basic.ForegroundUpdateService
import org.breezyweather.common.basic.models.Location
import org.breezyweather.remoteviews.Notifications
import org.breezyweather.remoteviews.presenters.notification.ForecastNotificationIMP

/**
 * Foreground Today forecast update service.
 */
@AndroidEntryPoint
class ForegroundTodayForecastUpdateService : ForegroundUpdateService() {
    override fun updateView(context: Context, location: Location) {
        if (ForecastNotificationIMP.isEnabled(this, true)) {
            ForecastNotificationIMP.buildForecastAndSendIt(context, location, true)
        }
    }

    override fun updateView(context: Context, locationList: List<Location>) {}
    override fun handlePollingResult(failed: Boolean) {
        // do nothing.
    }

    override fun getForegroundNotification(total: Int): NotificationCompat.Builder {
        return super.getForegroundNotification(total).setContentTitle(
            getString(R.string.breezy_weather) + " " + getString(R.string.notification_channel_forecast)
        )
    }

    override val foregroundNotificationId = Notifications.ID_UPDATING_TODAY_FORECAST
}